### DISCUSSION TITLE
How do you build the Apple Metal PJRT plugin?

### DISCUSSION THREAD

**Discussion by joelberkeley** [jax-ml/jax#25077] [Q&A]
Hi, I want to build or download a PJRT plugin for Apple Metal for a non-Python frontend (for all intents and purposes I'm working in C/C++). Do you know how to do this? I've seen [Apple's JAX Metal webpage](https://developer.apple.com/metal/jax/) but it's Python-specific and closed-source.

**Comment by jakevdp** [comment]
The apple metal plugin is closed-source code owned by Apple, so I don't think it's possible to build it yourself. CC/ @shuhand0 @kulinseth who may have more information.

**Reply by shuhand0** [reply]
Yes, as @jakevdp replied, JAX-Metal is closed-sourced. JAX-Metal plugin package consists of several dynamic libraries, and they use PJRT C API as interface. So you may try to link your C/C++ applications to the library.

**Comment by joelberkeley** [comment]
Sorry I forgot about this, I've figured out how to extract it from the Python install and use it
